text/0000-extract-extender.md

```diff
@@ -0,0 +1,270 @@
+
+* We could deliver the extender via a separate tarball on the GitHub releases page, and as a separate image at `buildpacksio/extender:<version>`. In both cases, the directory structure would look like the following:
+
+```
+Permission     UID:GID       Size  Filetree                                   
+drwxr-xr-x         0:0      32 MB  └── cnb                                    
+drwxr-xr-x         0:0      32 MB      ├── lifecycle                                     
+-rwxrwxrwx         0:0        0 B      │   ├── extender
+-rw-r--r--         0:0      10 kB      │   ├── extender.sbom.cdx.json         
+-rw-r--r--         0:0      16 kB      │   ├── extender.sbom.spdx.json        
+-rw-r--r--         0:0      15 kB      │   ├── extender.sbom.syft.json
+-rw-r--r--         0:0      560 B      └── lifecycle.toml
+```
+
+* This should be accompanied in the lifecycle by a minor version bump, due to the breaking change in the delivery mechanism
+* When creating builders, pack could look for a `[lifecycle.extender]` table in [builder.toml](https://buildpacks.io/docs/reference/config/builder-config/#lifecycle-optional) to find the extender binary
+  * If `[lifecycle.extender]` is NOT present, pack could inspect the lifecycle version provided and pull the extender with the same version from the GitHub releases page
+    * This mimics the [behavior](https://github.com/buildpacks/pack/blob/a1edf2e9e3837c2e9eeca019b97c28896b40f778/pkg/client/create_builder.go#L437) when NO lifecycle version is specified (pack will pull the latest lifecycle that it knows about from the GitHub releases page)
+  * Alternatively, we could look for a sibling tarball or image for the lifecycle provided (e.g., if provided `buildpacksio/lifecycle:<version>` we could look for `buildpacksio/extender:<version>`), but this seems risky as it is non-declarative
+  * Alternatively, we could put the lifecycle and the extender in separate layers within the same image
+* pack should put `/cnb/lifecycle/lifecycle` and `/cnb/lifecycle/extender` within the SAME "lifecycle layer" in the builder image
+  * We'd need to be careful not to write the parent directory (`/cnb/lifecycle`) twice, as this can produce an invalid image
+  * From a platform's perspective, nothing about the builder would have really changed, except that the extender is no longer a symlink
+
+* When running builds, platforms would launch extend-build and extend-run containers
+  * Using the builder image (as before) when doing build image extension
+  * Using the run image + extender binary from the _extender_ image (this is different) when doing run image extension
+    * See [here](https://github.com/buildpacks/pack/blob/a1edf2e9e3837c2e9eeca019b97c28896b40f778/pkg/client/build.go#L662) for how this is working today in pack
+    * Other platforms would need to make a similar change in order to upgrade to a newer lifecycle version IF they are looking in the lifecycle image for the extender binary (vs the builder)
```

Comment:
Do such platforms exist? Anecdotally the support requests I've seen re: run image extension (vs switching) were using `pack build`